File: atcoder.jp/contests/abc145/tasks/abc145_b/main.hs
yesNo :: Bool -> String
yesNo False = "No"
yesNo True = "Yes"

main :: IO ()
main = do
  n <- readLn
  s <- getLine

  let m = n `div` 2
      ans = take m s == drop m s

  putStrLn $ yesNo ans

File: atcoder.jp/contests/abc145/tasks/abc145_b/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  int N;
  string S;
  cin >> N >> S;

  int m = N / 2;
  bool ans = S.substr(0, m) == S.substr(m);

  cout << (ans ? "Yes" : "No") << endl;
}
